src/api/Nhs.Appointments.Core/AvailabilityService.cs

```diff
@@ -2,7 +2,7 @@
 
 public class AvailabilityService(IAvailabilityStore availabilityStore, IAvailabilityCreatedEventStore availabilityCreatedEventStore) : IAvailabilityService
 {
-    public async Task ApplyAvailabilityTemplateAsync(string site, DateOnly from, DateOnly until, Template template, string user)
+    public async Task ApplyAvailabilityTemplateAsync(string? site, DateOnly from, DateOnly until, Template? template, string user)
```

Comment:
These sorts of warnings I'd rather resolve by turning on TreatWarningsAsErrors then leaving the type non-nullable, but at the moment that would cause hundreds of changes in the codebase. 

We're checking for null a few lines down, so to avoid a warning the type currently needs to declare itself as potentially null. I think it all comes down to making the intention clear in the method signature

Reply:
I'm not a fan of this I would prefer to turn off nullable reference types